core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -702,8 +712,6 @@ private[deploy] class Master(
     worker.addExecutor(exec)
     worker.endpoint.send(LaunchExecutor(masterUrl,
       exec.application.id, exec.id, exec.application.desc, exec.cores, exec.memory))
-    exec.application.driver.send(ExecutorAdded(
-      exec.id, worker.id, worker.hostPort, exec.cores, exec.memory))
```

Comment:
I think we should keep this one. The idea of `ExecutorAdded` is to notify the application that the master has launched an executor for it. Then once the executor starts running the master will send an `ExecutorUpdated` message to the application, and we already do this when the master receives `ExecutorStateChanged`.

It's probably not a big deal either way but it's probably a better use of the different states.